src/engine/runtime.js

```diff
@@ -1219,10 +1219,24 @@ class Runtime extends EventEmitter {
         // Sanitize the placeholder to ensure valid XML
         placeholder = placeholder.replace(/[<"&]/, '_');
 
-        const argJSON = {
-            type: 'input_value',
-            name: placeholder
-        };
+        const iconURI = context.blockInfo.arguments[placeholder].iconURI;
```

Comment:
This code should probably go after the code below (https://github.com/LLK/scratch-vm/pull/2259/files#diff-38d875e1d50b3158dc96404b65275a54R1241) that declares `argInfo` because it handles the case where `context.blockInfo.arguments[placeholder]` does not exist. As is right now, this line of code will throw an error if that doesn't exist.

Reply:
I am also a little concerned about this approach to specifying an image field from an extension spec design perspective in general. Whether or not `field_image` is used in place of `input_value` should be determined by the argument type as opposed to whether a regular argument has an additional `iconURI` property provided.